core/src/main/java/org/bptlab/cepta/patterns/StaysInStationPattern.java

```diff
@@ -0,0 +1,67 @@
+package org.bptlab.cepta.patterns;
+
+import org.apache.flink.streaming.api.windowing.time.Time;
+import org.apache.flink.streaming.api.datastream.DataStreamUtils;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.cep.PatternStream;
+import org.apache.flink.cep.CEP;
+import org.apache.flink.cep.pattern.*;
+import org.apache.flink.cep.pattern.conditions.*;
+import org.bptlab.cepta.models.events.train.LiveTrainDataProtos.LiveTrainData;
+import org.bptlab.cepta.models.events.correlatedEvents.StaysInStationEventProtos.StaysInStationEvent;
+
+import java.util.*;
+
+
+public class StaysInStationPattern {   
+    public static final Pattern<LiveTrainData, ?> staysInStationPattern = 
+      Pattern.<LiveTrainData>begin("arrivesInStation")
+      .where(new SimpleCondition<LiveTrainData>(){
+        @Override
+        public boolean filter(LiveTrainData event) {
+          return event.getStatus() == 3;
+        }
+      })
+      .next("departuresFromStation")
```

Comment:
I would recommend using `followedBy` instead of `next`. Then it wouldn't only match on directly neighboring events.

Reply:
Yo